src/epics/explorer/deleteItem.epic.ts

```diff
@@ -14,30 +14,58 @@
 // You should have received a copy of the GNU General Public License
 // along with Superblocks Lab.  If not, see <http://www.gnu.org/licenses/>.
 
-import { switchMap } from 'rxjs/operators';
+import { catchError, switchMap } from 'rxjs/operators';
 import { ofType, Epic } from 'redux-observable';
-import { explorerActions, panesActions } from '../../actions';
+import { explorerActions, panesActions, projectsActions } from '../../actions';
 import { projectSelectors } from '../../selectors';
 import { projectService } from '../../services';
 import { updateItemInTree } from '../../reducers/explorerLib';
+import { fetchJSON } from '../../services/utils/fetchJson';
+import { of } from 'rxjs';
 
 export const deleteItemEpic: Epic = (action$, state$) => action$.pipe(
     ofType(explorerActions.DELETE_ITEM),
     switchMap((action) => {
         const project = projectSelectors.getProject(state$.value);
         const explorerState = state$.value.explorer;
+        const isOwnProject = state$.value.projects.isOwnProject;
 
-        return projectService.putProjectById(project.id, {
-            name: project.name,
-            description: project.description,
-            // TODO: remove usage of updateItemInTree. This is done only temporary.
-            files: updateItemInTree(explorerState.tree, explorerState.lastDeletedId, () => null)[0]
-        })
-        .pipe(
-            switchMap(() => {
-                return [panesActions.closePane(action.data.id), explorerActions.deleteItemSuccess(action.data.id)];
+        if (isOwnProject) {
+            return projectService.putProjectById(project.id, {
+                name: project.name,
+                description: project.description,
+                // TODO: remove usage of updateItemInTree. This is done only temporary.
+                files: updateItemInTree(explorerState.tree, explorerState.lastDeletedId, () => null)[0]
             })
-            // TODO: error handling
-        );
+                .pipe(
+                    switchMap(() => {
+                        return [panesActions.closePane(action.data.id), explorerActions.deleteItemSuccess(action.data.id)];
+                    }),
+                    catchError(() => [explorerActions.deleteItemFail(action.data.id)])
+                );
+        } else {
+            // fork with new tree structure
+            return projectService.createProject({
```

Comment:
This part of code seems to be repeated all over in every action. Please refactor it out in separate epic and just produce some action to trigger that epic if possible.